src/NuGet.Core/NuGet.Commands/SourcesCommands/SourceRunners.cs

```
@@ -336,6 +336,27 @@ public static ISettings GetSettings(string configfile)
             }
         }
 
+        public static ISettings GetSettingsFromDirectory(string directory)
+        {
+            if (Directory.Exists(directory))
+            {
+                return NuGet.Configuration.Settings.LoadDefaultSettings(directory,
+                   configFileName: null,
+                   machineWideSettings: new XPlatMachineWideSetting());
+            }
+            else if (string.IsNullOrEmpty(directory))
+            {
+                var currentDirectory = Directory.GetCurrentDirectory();
+                return NuGet.Configuration.Settings.LoadDefaultSettings(currentDirectory,
+                   configFileName: null,
+                   machineWideSettings: new XPlatMachineWideSetting());
+            }
+            else
+            {
+                return null;
```

I'm not sure exactly if this method should return `null` or just throw the exception that the caller throws if the specified directory does not exist.  What does everyone think is cleaner?

---

From the [spec](https://github.com/NuGet/Home/blob/dev/proposed/2022/dotnet-nuget-config-spec.md#arguments),
`If the specified WORKING_DIRECTORY doesn't exist, an error is displayed indicating the WORKING_DIRECTORY doesn't exist.`
The [example ](https://github.com/NuGet/Home/blob/dev/proposed/2022/dotnet-nuget-config-spec.md#examples)in the spec:
```
List all the NuGet configuration file that will be applied, but passing a non-existing argument WORKING_DIRECTORY.
dotnet nuget config paths C:\Test\NonExistingRepos

Error: The path "C:\Test\NonExistingRepos" doesn't exist.
```

---

Martin and I discussed this, and throwing the exception instead of returning null made more sense to us.